### PR TITLE
support "Not Stated" options for boolean options and provide appropri…

### PR DESCRIPTION
**Pull request by dsnyder-fairway**
…ate translations for undefined subgroups

I removed Strict boolean related classes as we no longer are using this

### CODE REVIEW

**Comment by jtreuting** [comment]
@dsnyder-fairway Does this add the 3rd option to all the forms that previously only had 2?  Like advanced filters and customize subgroups?  I'm not sure if we want that in all places which is why i'm asking.

**Comment by dsnyder-fairway** [comment]
Yes, this allows users to select Not Stated for LEP, IEP, and Economic Disadvantage. We also return the correct translated string on the results

File: aggregate-service/src/main/java/org/opentestsystem/rdw/olap/service/impl/DefaultReportOptionService.java [hunk 1]
```diff
@@ -69,18 +65,17 @@
...
     public ReportOptions getReportOptions() {
         final List<String> booleans = codesOf(booleanRepository.findAll());
-        final List<String> strictBooleans = codesOf(strictBooleanRepository.findAll());
         return ReportOptions.builder()
                 .assessmentGrades(codesOf(assessmentGradeService.getAssessmentGrades()))
                 .completenesses(codesOf(completenessService.getCompletenesses()))
                 .claims(claimService.findAll())
                 .dimensionTypes(DimensionType.orderedDisplayValues())
-                .economicDisadvantages(strictBooleans)
+                .economicDisadvantages(booleans)
```
Comment (agorina) [inline aggregate-service/src/main/java/org/opentestsystem/rdw/olap/service/impl/DefaultReportOptionService.java]: I do not think this is what we want. @jtreuting , please correct me if needed, but I think we wanted to keep this as is.

Comment (jtreuting) [reply aggregate-service/src/main/java/org/opentestsystem/rdw/olap/service/impl/DefaultReportOptionService.java]: We don't want to be able to filter by not stated, just handle it on the display as a row in the results.

Comment (dsnyder-fairway) [reply aggregate-service/src/main/java/org/opentestsystem/rdw/olap/service/impl/DefaultReportOptionService.java]: fixing this pr now

File: webapp/src/main/webapp/src/app/aggregate-report/support.ts [hunk 2]
```diff
@@ -85,15 +85,19 @@ export function organizationOrdering<T>(organizationGetter: (item: T) => Organiz
 
 export function subgroupOrdering<T>(subgroupGetter: (item: T) => Subgroup, options: AggregateReportOptions): Ordering<T> {
 
+  function getNullableOption(option: string[]) {
+    return option.concat('undefined');
+  }
```
Comment (dsnyder-fairway) [inline webapp/src/main/webapp/src/app/aggregate-report/support.ts]: changes here are to correct the ordering when viewing the results

File: webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-summary.component.ts [hunk 3]
```diff
@@ -315,7 +315,7 @@ export class AggregateReportSummary {
           values: inline(orAll(
             optionFilters.individualEducationPlans,
             settingFilters.individualEducationPlans,
-            code => translate(`common.strict-boolean.${code}`)
+            code => translate(`common.boolean.${code}`)
```
Comment (dsnyder-fairway) [inline webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-summary.component.ts]: I don't think this has any impact either way, but I'm going to see what reverting this change does

**Comment by dsnyder-fairway** [comment]
I'll address any comments as they come

**Pull request merged**